test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/FormatterTest.scala

```diff
@@ -668,6 +668,27 @@ class FormatterTest {
         1, 1)
     expectFormatterThrows(classOf[MissingFormatArgumentException], "%10$d", 1)
   }
+
+  @Test def `BigInteger should be correctly formatted as a decimal string`(): Unit = {
+    val b = BigInt(new BigInteger("42"))
+    val result = b.formatted("%d")
```

Comment:
The tests should be inserted in the existing `should_provide_d_conversion` and friends, using the existing infrastructure (`assertF`) directly on `BigInteger`.

Also we should have some more tests:
* for negative numbers (and especially for negative numbers in octal and hexadecimal printing)
* with a few options, notably zero padding (`0`) and grouping (`,`)

You might also want to add a test in `FormatterTest.testFormatFrench()` and `testFormatHindiWithDevanagariDigits()`, to make sure that locale-sensitivity applies to formatted `BigInteger`s.